modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -1972,6 +1973,7 @@ private void handleReply(int[] operands, StackFrame sf) {
         if (i >= 0) {
             message = (BMessage) sf.refRegs[i];
         }
+        handleMessageHeaders(message);
```

Comment:
Since this method is only handling session, the name `handleMessageHeaders` is misleading.